src/App.tsx

```diff
@@ -73,45 +87,85 @@ class Board extends React.Component<BoardProps, BoardState> {
         });
     }
 
+    public cornerMarkSelectedSquares(i: SquareValue) {
+        const newMarkingMap = new Map(this.state.markingMap);
+        for (const address of this.state.highlights) {
+            if (!this.isPermanent(address) && this.state.values[address] == null) {
+                const marks = newMarkingMap.has(address) ? newMarkingMap.get(address) as SquareValue[] : [];
+                if (marks.includes(i)) {
+                    const index = marks.indexOf(i);
+                    marks.splice(index, 1);
+                } else {
+                    marks.push(i);
+                }
+                newMarkingMap.set(address, marks);
+            }
+        }
+        this.setState({
+            ...this.state,
+            markingMap : newMarkingMap,
+        });
+    }
+
     /**
      * onClick for DELETE button
      */
     public deleteSelectedSquares(): void {
         const newValues = JSON.parse(JSON.stringify(this.state.values));
+        const newMarkingMap = new Map(this.state.markingMap);
+        let deleteValues: boolean = false; // Determine weather to delete all values or delete all markings
+        for (const address of this.state.highlights) {
+            if (!this.isPermanent(address) && newValues[address] != null) {
+                deleteValues = true;
+            }
+        }
         for (const address of this.state.highlights) {
```

Comment:
Consolidate these two loops into one loop, remove the `deleteValues` variable.

Reply:
I do not think it can be done. If there are 4 squares highlighted and 2 only have markings and 2 have user filled values. The first time you click delete I only want the 2 values to be deleted and the second click would delete the markings. So you must iterate over the highlights once to check if any of the squares have a user filled value and then a second time to accordingly delete the Square contents.

Reply:
Makes sense. In this case, I would add a `break` after 136 above.